milli/src/search/query_tree.rs

```diff
@@ -857,532 +852,263 @@ mod test {
         let query = "hey friends";
         let tokens = query.tokenize();
 
-        let expected = Operation::Or(
-            false,
-            vec![
-                Operation::And(vec![
-                    Operation::Query(Query {
-                        prefix: false,
-                        kind: QueryKind::exact("hey".to_string()),
-                    }),
-                    Operation::Query(Query {
-                        prefix: true,
-                        kind: QueryKind::tolerant(1, "friends".to_string()),
-                    }),
-                ]),
-                Operation::Query(Query {
-                    prefix: true,
-                    kind: QueryKind::tolerant(1, "heyfriends".to_string()),
-                }),
-            ],
-        );
-
         let (query_tree, _) =
             TestContext::default().build(false, true, None, tokens).unwrap().unwrap();
 
-        assert_eq!(expected, query_tree);
+        insta::assert_debug_snapshot!(query_tree, @r###"
+        OR
+          AND
+            Exact { word: "hey" }
+            PrefixTolerant { word: "friends", max typo: 1 }
+          PrefixTolerant { word: "heyfriends", max typo: 1 }
```

Comment:
We must make sure that this debug-printing is reliable and that the whole set of fields is displayed.

Reply:
Yep, I checked it :)